### ISSUE TITLE
Request for pre-trained models.

### ISSUE THREAD

**Issue by akshitjindal1** [HanxunH/Detect-CLIP-Backdoor-Samples#1]
Hello,
Congratulations on the paper acceptance!
Could you share the pretrained models that you used for experimentation?
It would be a great help as I lack the resources to perform the pre-training step.

**Comment by HanxunH** [comment]
> Hello, Congratulations on the paper acceptance! Could you share the pretrained models that you used for experimentation? It would be a great help as I lack the resources to perform the pre-training step.

I’ll get back to you next week. Is there a particular backdoored model you’re interested in?

**Comment by akshitjindal1** [comment]
I would like to have the STBA models for Vit and RN50 as you have presented in table 1 of your paper.

**Comment by HanxunH** [comment]
> I would like to have the STBA models for Vit and RN50 as you have presented in table 1 of your paper.

Hi,

Thanks for your interest! You can find the checkpoints here: https://drive.google.com/drive/folders/1yxogWGnrd_YEyg0BgxlR-h-lg3fBkphL?usp=sharing

I have also uploaded a QuickStart notebook and updated the README to guide you through using the pre-trained weights with this notebook. This may be helpful as well.

**Comment by akshitjindal1** [comment]
The model weights that you have provided have almost zero accuracy on clean imagenet samples. I used your quickstart notebook to write my own code for model loading and testing and every model has around 0.2% accuracy on clean imagenet data. Can you verify this at your end?

```
import torch
import torchvision.datasets as datasets
import torchvision.transforms as transforms
from torch.utils.data import DataLoader, Subset
import numpy as np
from functools import partial
import models.clip_model
import models.modified_resnet
import models.transformer
from open_clip import get_tokenizer
import torch.nn.functional as F
from tqdm import tqdm

def test_model(model_path, imagenet_path, device='cuda:1'):
    # Load model
    checkpoint = torch.load(model_path, map_location=device)
    state_dict = checkpoint.get("state_dict", checkpoint)
    if next(iter(state_dict.items()))[0].startswith("module"):
        state_dict = {key[len("module."):]: value for key, value in state_dict.items()}

    # Initialize model
    vision_model_config = partial(
        models.modified_resnet.ModifiedResNet,
        layers=[3,4,6,3],
        heads=32,
        width=64,
        image_size=224,
        output_dim=1024,
        get_features=True,
    )

    text_model_config = partial(
        models.transformer.TextTransformer,
        context_length=77,
        vocab_size=49408,
        width=512,
        heads=8,
        layers=12,
        output_dim=1024
    )

    model = models.clip_model.CLIP(vision_model_config, text_model_config)
    model.load_state_dict(state_dict)
    model = model.to(device)
    model.eval()

    # Prepare data and tokenizer
    transform = transforms.Compose([
        transforms.Resize(224, interpolation=transforms.InterpolationMode.BICUBIC),
        transforms.CenterCrop(224),
        transforms.ToTensor(),
        transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
    ])

    dataset = datasets.ImageNet(root=imagenet_path, split='val', transform=transform)
    indices = np.random.choice(len(dataset), size=1000, replace=False)
    subset = Subset(dataset, indices)
    loader = DataLoader(subset, batch_size=64, num_workers=4)

    tokenizer = get_tokenizer('RN50')
    templates = ["a photo of a {}"]
    classnames = dataset.classes

    with torch.no_grad():
        # Get text features
        zeroshot_weights = []
        for classname in tqdm(classnames, desc="Processing classes"):
            texts = [template.format(classname) for template in templates]
            text_tokens = tokenizer(texts).to(device)
            text_features = model.encode_text(text_tokens)
            text_features = text_features.mean(dim=0)
            text_features /= text_features.norm()
            zeroshot_weights.append(text_features)
        
        zeroshot_weights = torch.stack(zeroshot_weights, dim=1).to(device)

        # Evaluate
        correct = 0
        total = 0
        for images, labels in tqdm(loader, desc="Testing"):
            images = images.to(device)
            labels = labels.to(device)

            image_features = model.encode_image(images, normalize=True)
            logits = model.logit_scale.exp() * image_features @ zeroshot_weights
            predictions = logits.argmax(dim=1)
            
            correct += (predictions == labels).sum().item()
            total += labels.size(0)

        accuracy = 100 * correct / total
        print(f"Top-1 Accuracy: {accuracy:.2f}%")

if __name__ == "__main__":
    test_model(
        model_path="rn50_wanet_cc3m.pt",
        imagenet_path='datasets/Imagenet_full/',
        device='cuda:1'
    )
```

**Comment by HanxunH** [comment]
The clean accuracy should be approximately 17%. These weights were trained without input image normalization. Try removing the following normalization transform and see if it works: `transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])`

**Comment by akshitjindal1** [comment]
That works, thanks. Closing this thread.
